packages/hawtio/src/plugins/runtime/SysProps.tsx

```diff
@@ -0,0 +1,235 @@
+  ToolbarGroup,
+  ToolbarItem,
+} from '@patternfly/react-core'
+import { TableComposable, Tbody, Td, Th, Thead, ThProps, Tr } from '@patternfly/react-table'
+import { SearchIcon } from '@patternfly/react-icons'
+import { getSystemProperties } from './runtime-service'
+import { objectSorter } from '@hawtiosrc/util/objects'
+import { SystemProperty } from './types'
+
+export const SysProps: React.FunctionComponent = () => {
+  const [properties, setProperties] = useState<{ key: string; value: string }[]>([])
+  const [filteredProperties, setFilteredProperties] = useState<SystemProperty[]>([])
+  const [page, setPage] = useState(1)
+  const [perPage, setPerPage] = useState(20)
+  const [searchTerm, setSearchTerm] = useState('')
+  const [filters, setFilters] = useState<string[]>([])
+  const [attributeMenuItem, setAttributeMenuItem] = useState('name')
+  const [sortIndex, setSortIndex] = React.useState<number>(-1)
+  const [sortDirection, setSortDirection] = React.useState<'asc' | 'desc'>('asc')
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false)
+
+  useEffect(() => {
+    getSystemProperties().then(props => {
+      setProperties(props)
+      setFilteredProperties(props)
+    })
+  }, [])
+
+  const onDeleteFilter = (filter: string) => {
+    const newFilters = filters.filter(f => f !== filter)
+    setFilters(newFilters)
+    handleSearch(searchTerm, attributeMenuItem, newFilters)
+  }
+
+  const addToFilters = () => {
+    setFilters([...filters, `${attributeMenuItem}:${searchTerm}`])
+    setSearchTerm('')
+  }
+  const clearFilters = () => {
+    setFilters([])
+    setSearchTerm('')
+    handleSearch('', attributeMenuItem, [])
+  }
+
+  const PropsPagination = () => {
+    return (
+      <Pagination
+        itemCount={filteredProperties.length}
+        page={page}
+        perPage={perPage}
+        onSetPage={(_evt, value) => setPage(value)}
+        onPerPageSelect={(_evt, value) => {
+          setPerPage(value)
+          setPage(1)
+        }}
+        variant='top'
+      />
+    )
+  }
+
+  const getPageProperties = (): SystemProperty[] => {
+    const start = (page - 1) * perPage
+    const end = start + perPage
+    return filteredProperties.slice(start, end)
+  }
+  const handleSearch = (value: string, attribute: string, filters: string[]) => {
+    setSearchTerm(value)
+    //filter with findTerm
+    let filtered: SystemProperty[] = []
+
+    if (value === '') {
+      filtered = [...properties]
+    } else {
+      filtered = properties.filter(prop => {
+        return (attribute === 'name' ? prop.key : prop.value).toLowerCase().includes(value.toLowerCase())
+      })
+    }
+
+    //filter with the rest of the filters
+    filters.forEach(value => {
+      const attr = value.split(':')[0] ?? ''
+      const searchTerm = value.split(':')[1] ?? ''
+      filtered = filtered.filter(prop =>
+        (attr === 'name' ? prop.key : prop.value).toLowerCase().includes(searchTerm.toLowerCase()),
+      )
+    })
```

Comment:
The filtering logic itself is something we can move to runtime-service.ts, so that the component logic can be simplified.

Reply:
hmm I was rather thinking to create a new generic table component with sorting and filtering at some point. Because I have sort of similar things at 3 places.

Reply:
Sounds good. Then maybe we can still put the filtering logic to a generic table-service.ts.

But this is not a big problem so we can leave it as is for now if opinions diverge.